json: cgen error with array of struct pointers []&T

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

**V doctor:**

```
V full version: V 0.4.8 844d89f.f2c281e
OS: macos, macOS, 15.1.1, 24B2091
Processor: 10 cpus, 64bit, little endian, Apple M4

getwd: /Users/leodev/v
vexe: /Users/leodev/v2/v
vexe mtime: 2024-11-29 18:18:54

vroot: OK, value: /Users/leodev/v2
VMODULES: OK, value: /Users/leodev/.vmodules
VTMP: OK, value: /tmp/v_501

Git version: git version 2.47.0
Git vroot status: weekly.2024.48-22-gf2c281ea
.git/config present: true

CC version: Apple clang version 16.0.0 (clang-1600.0.26.4)
emcc version: N/A
thirdparty/tcc status: thirdparty-macos-arm64 713692d4
```

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg repro.v`

```v
import json

struct Test {
	a string
}

fn main() {
	x := json.decode([]&Test, '[{"a":"a"}]') or {
		exit(1)
	}
	println(x)
}
```

**What did you expect to see?**

Successful compilation

**What did you see instead?**

```
================== C compilation error (from cc): ==============
cc: /tmp/v_501/repro.01JDWJBSA1PV3RS9PP1PFTB8PA.tmp.c:2347:3: error: unknown type name '_result_main__Test'; did you mean '_result_main__Test_ptr'?
cc:  2347 |                 _result_main__Test* val2 = json__decode_main__Test_ptr ((cJSON *)jsval);
cc:       |                 ^~~~~~~~~~~~~~~~~~
cc:       |                 _result_main__Test_ptr
cc: /tmp/v_501/repro.01JDWJBSA1PV3RS9PP1PFTB8PA.tmp.c:85:39: note: '_result_main__Test_ptr' declared here
cc:    85 | typedef struct _result_main__Test_ptr _result_main__Test_ptr;
cc:       |                                       ^
cc: /tmp/v_501/repro.01JDWJBSA1PV3RS9PP1PFTB8PA.tmp.c:2347:23: error: initializing '_result_main__Test_ptr *' (aka 'struct _result_main__Test_ptr *') with an expression of incompatible type '_result_main__Test_ptr' (aka 'struct _result_main__Test_ptr')
cc:  2347 |                 _result_main__Test* val2 = json__decode_main__Test_ptr ((cJSON *)jsval);
cc:       |                                     ^      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc: /tmp/v_501/repro.01JDWJBSA1PV3RS9PP1PFTB8PA.tmp.c:2348:10: error: member reference type '_result_main__Test_ptr *' (aka 'struct _result_main__Test_ptr *') is a pointer; did you mean to use '->'?
cc:  2348 |                 if(val2.is_error) {
... (the original output was 19 lines long, and was truncated to 12 lines)
================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

<sub><a href="https://huly.app/guest/vlang-66f40c4d-a476b54c67-771fdd?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2NzRhMDVjNGQyZWY0Y2JjYzQ2Y2E3ZmIiLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6InctYWxleGFuZGVyLXZsYW5nLTY2ZjQwYzRkLWE0NzZiNTRjNjctNzcxZmRjIn0.XwlJlQC2NeGr6RBJ-WZfWOC8_eIYnnEhZrr1t2PIQKo">Huly&reg;: <b>V_0.6-21448</b></a></sub>

--- a/vlib/json/tests/json_decode_arr_ref_test.v
+++ b/vlib/json/tests/json_decode_arr_ref_test.v
@@ -1,0 +1,10 @@
+import json
+
+struct Test {
+	a string
+}
+
+fn test_main() {
+	x := json.decode([]&Test, '[{"a":"a"}]') or { exit(1) }
+	assert x[0].a == 'a'
+}

--- a/vlib/v/gen/c/json.v
+++ b/vlib/v/gen/c/json.v
@@ -1021,7 +1021,7 @@ fn (mut g Gen) decode_array(utyp ast.Type, value_type ast.Type, fixed_array_size
 		s = '${styp} val = ${fn_name}((cJSON *)jsval); '
 	} else if is_array_fixed_val {
 		s = '
-		${result_name}_${styp} val2 = ${fn_name} ((cJSON *)jsval);
+		${result_name}_${styp.replace('*', '_ptr')} val2 = ${fn_name} ((cJSON *)jsval);
 		if(val2.is_error) {
 			${array_free_str}
 			return *(${result_name}_${ret_styp}*)&val2;
@@ -1030,7 +1030,7 @@ fn (mut g Gen) decode_array(utyp ast.Type, value_type ast.Type, fixed_array_size
 		memcpy(&val, (${styp}*)val2.data, sizeof(${styp}));'
 	} else {
 		s = '
-		${result_name}_${styp} val2 = ${fn_name} ((cJSON *)jsval);
+		${result_name}_${styp.replace('*', '_ptr')} val2 = ${fn_name} ((cJSON *)jsval);
 		if(val2.is_error) {
 			${array_free_str}
 			return *(${result_name}_${ret_styp}*)&val2;

